lib/server.js

```diff
@@ -44,196 +39,200 @@ module.exports = Server
+  _runWorker (queueName, job, done) {
+    debug('_runWorker(' + [ queueName, job ].join(', ') + ')')
+    let opts = clone(this._workerOptions[queueName])
+    defaults(opts, {
+      queue: queueName,
+      job: job,
+      task: this._tasks[queueName],
+      done: done,
+      log: this.log,
+      errorCat: this.errorCat
     })
-}
+    Worker.create(opts)
+  }
 
-/**
- * Stops the worker server.
- * @return {promise} A promise that resolves when the server is stopped.
- */
-Server.prototype.stop = function () {
-  debug('stop')
-  this.log.trace('ponos.Server: stopping')
-  return this._unsubscribeAll().bind(this)
-    .then(function () {
-      return this.hermes.closeAsync()
-    })
-    .then(function () {
-      this.log.trace('ponos.Server: stopped')
-    })
-    .catch(function (err) {
-      this.errorCat.report(err)
-      throw err
-    })
-}
+  /**
+   * Starts the worker server and listens for jobs coming from all queues.
+   * @return {promise} A promise that resolves when the server is listening.
+   */
+  start () {
+    debug('start')
+    this.log.trace('ponos.Server: starting')
+    return this.hermes.connectAsync().bind(this)
+      .then(this._subscribeAll)
+      .then(() => {
+        this.log.trace('ponos.Server: started')
+      })
+      .catch((err) => {
+        this.errorCat.report(err)
+        throw err
+      })
+  }
 
-/**
- * Takes a map of queues and task handlers and sets them all.
- * @param {object} map A map of queue names to task handlers.
- * @param {string} map.key Queue name.
- * @param {function} map.value Function to take a job or an object with a task
- *   and additional options for the worker.
- * @returns {ponos.Server} The server.
- */
-Server.prototype.setAllTasks = function (map) {
-  debug('setAllTasks')
-  Object.keys(map).forEach(function (key) {
-    var value = map[key]
-    if (isObject(value)) {
-      if (!isFunction(value.task)) {
-        log.warn({
-          key: key
-        }, 'ponos.Server.setAllTasks: No task function defined')
-        return
+  /**
+   * Stops the worker server.
+   * @return {promise} A promise that resolves when the server is stopped.
+   */
+  stop () {
+    debug('stop')
+    this.log.trace('ponos.Server: stopping')
+    return this._unsubscribeAll().bind(this)
+      .then(() => {
+        return this.hermes.closeAsync()
+      })
+      .then(() => {
+        this.log.trace('ponos.Server: stopped')
+      })
+      .catch((err) => {
+        this.errorCat.report(err)
+        throw err
+      })
+  }
+
+  /**
+   * Takes a map of queues and task handlers and sets them all.
+   * @param {object} map A map of queue names to task handlers.
+   * @param {string} map.key Queue name.
+   * @param {function} map.value Function to take a job or an object with a task
+   *   and additional options for the worker.
+   * @returns {ponos.Server} The server.
+   */
+  setAllTasks (map) {
+    debug('setAllTasks')
+    Object.keys(map).forEach(function (key) {
```

Comment:
We should probably check if `isObject(map)` here to avoid the unpleasantness of a potential thrown error. We could simply bypass the logic and log a warning or error if the end user accidentally attempts to pass a value that cannot be "keyed".